### PR TITLE
Enhance shell detection in `env` command

### LINKED ISSUE #1441: docker-machine on Windows / msysgit bash cannot generate env keys (v1.3)
Tried the normal `eval "$(docker-machine env dev)"`, but that resulted in an error that I should pass `--shell` with either `cmd` or `powershell`. 

I tried to pass `bash` (and `powershell`), but both attempts hung. I did get the output when I passed `cmd` and could manually set the values (via `export`).

### PR DESCRIPTION
**Pull request by jeanlaurent**
This fix : 
- un-hide the message displayed to the user on windows stating to use the `--shell` flag
- On windows, default to cmd if we detect we are not running in cywin/mingw etc...
- Properly detect fish shell
- Unit test for everyone.

This fixes #1441, #732

### CODE REVIEW

**Comment by jeanlaurent** [comment]
Ready for review cc @docker/machine-maintainers

File: commands/env_windows_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDetect(t *testing.T) {
+	originalShell := os.Getenv("SHELL")
+	os.Setenv("SHELL", "")
+	defer os.Setenv("SHELL", originalShell)
+	shell, _ := detectShell()
```
Comment (nathanleclaire) [inline commands/env_windows_test.go]: Can we check the error please? :wink:

File: commands/env_test.go [hunk 2]
```diff
@@ -547,3 +547,22 @@ func TestShellCfgUnset(t *testing.T) {
 		os.Setenv(test.noProxyVar, "")
 	}
 }
+
+func TestDetectBash(t *testing.T) {
+	originalShell := os.Getenv("SHELL")
+	os.Setenv("SHELL", "/bin/bash")
+	defer os.Setenv("SHELL", originalShell)
+	shell, _ := detectShell()
```
Comment (nathanleclaire) [inline commands/env_test.go]: Also here and in that other test.

**Comment by nathanleclaire** [comment]
Please check the errors in the tests and then LGTM

**Comment by dgageot** [comment]
LGTM when the tests are fixed

**Comment by jeanlaurent** [comment]
Integrated the tests check

**Pull request merged**